src/tests/ftest/recovery/pool_list_consolidation.py

```diff
@@ -290,16 +290,23 @@ def test_lost_majority_ps_replicas(self):
                 count += 1
                 if count > 1:
                     break
-
-        # 4. Start servers.
+        if count == 0:
```

Comment:
It's probably safer to also check `self.server_managers[0].manager.job.using_control_metadata` so we don't accidentally skip this for non-md-on-ssd.

And similar for other changes. What if `check_file_exists` doesn't find the file on a pmem cluster?

Reply:
I updated to also check `using_control_metadata`.
I think we can keep the usage of `check_file_exists` or `using_control_metadata` for the purpose of detecting the type of cluster (MD-on-SSD vs. PMEM) and we should assume that the file/directory is created properly. For example, in `test_lost_all_rdb`, let's say `rdb-pool` isn't created in the first place for non-md-on-ssd cluster. This means DAOS is broken, so we should fail the test. To support this feature, we should first check that `rdb-pool` exists in a non-md-on-ssd cluster, but I think we can assume that `rdb-pool` exists because creating it during pool create isn't related to CR.